pkg/roachprod/roachprod.go

```diff
@@ -724,8 +735,8 @@ func Extend(l *logger.Logger, clusterName string, lifetime time.Duration) error
 		return err
 	}
 
-	// Reload the clusters and print details.
-	c, err = getClusterFromCloud(l, clusterName)
+	// Save the cluster to the cache and print details.
+	err = saveCluster(l, c)
```

Comment:
I think this is thread-safe since no two threads would be touching the same `clusterName` and the filename is uniquely determined. It's worth double-checking since roachtest workers are manipulating cluster state(s), concurrently.

Reply:
It is thread-safe. `saveCluster` creates a temporary file, then renames it.